2022/incoming/moosex-getopt-saves-christmas.pod

```diff
@@ -0,0 +1,149 @@
+Author: Mike Whitaker <mike@altrion.org>
+Title: MooseX::Getopt saves Christmas
+Topic: getopt, moose
+
+=encoding utf8
+
+=head1 MooseX::Getopt saves Christmas
+
+"It's done, sir." An elf that had the fuzzy, out of focus look of a developer who'd been awake since the beginning of Advent stuck their head round Santa's door. "New, uh, present delivery planning script, like you asked. No more gender-specific toys, and naughty or nice checks are now optional. And it feeds the new delivery plotting script or your SledNav."
+
+"Hrm. OK." Santa nodded, distractedly. "I'll take a look later."
+
+They blinked. "OK. I'll just... uh... go crash for a bit." Looking like they 
+would do so before they made it much further down the corridor, they stumbled out.
+
+=head2 Some while later:
+
+Santa brought up a shell prompt. "Hrmmm. Let's see. Ahah."
+
+    $ delivery_plan
+    Usage: delivery_plan [options]
+
+Saint Nick harumphed.
+
+    $ delivery_plan --help
+    Usage: delivery_plan [options]
+
+"ERNEST!"
+
+Santa's right-hand elf hurried in. "Yes boss?"
+
+"What in the name of Christmas am I supposed to do to get this to run?"
+
+Ernest peered over Santa's shoulder. "Is that the new delivery script? I better take a look - the dev is snoring fit to be heard in Bethlehem, and I don't think they'll wake up before New Year."
+
+=head2 Command line options
+
+Ernest sighed, back at his desk. "OK. No marks for hand-rolling the option parsing..." He copied the main script to C<NorthPole/DeliveryPlan.pm> and edited it, removing the crufty command line parsing code and wrapping the remainder of the body of the script in a method.
+
+=begin perl
+
+    package NorthPole::DeliveryPlan;
+    use 5.36;
+    use Moose;
+    with MooseX::Getopt;
+
+    use feature 'signatures';
+
+    sub run ($self) {
+        # body of script
+    }
+
+=end perl
+
+Santa's right-hand elf's right-hand elf peered curiously over Ernest's shoulder. "Whatcha doing?" she asked.
```

Comment:
```suggestion
Santa's right-hand elf's peered curiously over Ernest's shoulder. "Whatcha doing?" she asked.
```

Reply:
Er no, Ernest is Santa's right hand elf, she's Ernest's right hand elf, which makes her "Santa's right-hand elf's right-hand elf" :D

Reply:
Ah OK, sorry sir :D